### PR TITLE
torch.sparse.softmax avoid div by zero and invalid kernel launch parameters

### PR DESCRIPTION
**Pull request by cpuhrsch**
### Description
Small changes needed to deal with nnz 0 inputs.

### Issue
https://github.com/pytorch/pytorch/issues/82107

### Testing
Added additional test coverage to reproduce bug reported in issue. Tested resulting values by conversion `to_dense`.

### CODE REVIEW

**Review by jbschlosser — APPROVED** [review]
LGTM

File: aten/src/ATen/native/sparse/cuda/SoftMax.cu [hunk 1]
```diff
@@ -408,7 +416,7 @@ void cuda_sparse_coo_softmax(
 
   auto nnz = values.size(0);
   auto sizes = input.sizes();
-  auto nvalues = values.numel() / nnz;
+  auto nvalues = get_nvalues(sizes, sparse_dim);
```
Comment (cpuhrsch) [inline aten/src/ATen/native/sparse/cuda/SoftMax.cu:419]: This is what the CPU kernel does.

**Review by ezyang — APPROVED** [review]
